option.go

```diff
@@ -38,65 +40,92 @@ func defaults(cfg *config) {
-	if _, ok := tags[constants.CIWorkspacePath]; !ok {
-		tags[constants.CIWorkspacePath] = gitData.SourceRoot
+	if _, ok := localTags[constants.CIWorkspacePath]; !ok {
+		localTags[constants.CIWorkspacePath] = gitData.SourceRoot
 	}
-	if _, ok := tags[constants.GitRepositoryURL]; !ok {
-		tags[constants.GitRepositoryURL] = gitData.RepositoryUrl
+	if _, ok := localTags[constants.GitRepositoryURL]; !ok {
+		localTags[constants.GitRepositoryURL] = gitData.RepositoryUrl
 	}
-	if _, ok := tags[constants.GitCommitSHA]; !ok {
-		tags[constants.GitCommitSHA] = gitData.CommitSha
+	if _, ok := localTags[constants.GitCommitSHA]; !ok {
+		localTags[constants.GitCommitSHA] = gitData.CommitSha
 	}
-	if _, ok := tags[constants.GitBranch]; !ok {
-		tags[constants.GitBranch] = gitData.Branch
+	if _, ok := localTags[constants.GitBranch]; !ok {
+		localTags[constants.GitBranch] = gitData.Branch
 	}
 
-	if tags[constants.GitCommitSHA] == gitData.CommitSha {
-		if _, ok := tags[constants.GitCommitAuthorDate]; !ok {
-			tags[constants.GitCommitAuthorDate] = gitData.AuthorDate.String()
+	if localTags[constants.GitCommitSHA] == gitData.CommitSha {
+		if _, ok := localTags[constants.GitCommitAuthorDate]; !ok {
+			localTags[constants.GitCommitAuthorDate] = gitData.AuthorDate.String()
 		}
-		if _, ok := tags[constants.GitCommitAuthorName]; !ok {
-			tags[constants.GitCommitAuthorName] = gitData.AuthorName
+		if _, ok := localTags[constants.GitCommitAuthorName]; !ok {
+			localTags[constants.GitCommitAuthorName] = gitData.AuthorName
 		}
-		if _, ok := tags[constants.GitCommitAuthorEmail]; !ok {
-			tags[constants.GitCommitAuthorEmail] = gitData.AuthorEmail
+		if _, ok := localTags[constants.GitCommitAuthorEmail]; !ok {
+			localTags[constants.GitCommitAuthorEmail] = gitData.AuthorEmail
 		}
-		if _, ok := tags[constants.GitCommitCommitterDate]; !ok {
-			tags[constants.GitCommitCommitterDate] = gitData.CommitterDate.String()
+		if _, ok := localTags[constants.GitCommitCommitterDate]; !ok {
+			localTags[constants.GitCommitCommitterDate] = gitData.CommitterDate.String()
 		}
-		if _, ok := tags[constants.GitCommitCommitterName]; !ok {
-			tags[constants.GitCommitCommitterName] = gitData.CommitterName
+		if _, ok := localTags[constants.GitCommitCommitterName]; !ok {
+			localTags[constants.GitCommitCommitterName] = gitData.CommitterName
 		}
-		if _, ok := tags[constants.GitCommitCommitterEmail]; !ok {
-			tags[constants.GitCommitCommitterEmail] = gitData.CommitterEmail
+		if _, ok := localTags[constants.GitCommitCommitterEmail]; !ok {
+			localTags[constants.GitCommitCommitterEmail] = gitData.CommitterEmail
 		}
-		if _, ok := tags[constants.GitCommitMessage]; !ok {
-			tags[constants.GitCommitMessage] = gitData.CommitMessage
+		if _, ok := localTags[constants.GitCommitMessage]; !ok {
+			localTags[constants.GitCommitMessage] = gitData.CommitMessage
 		}
 	}
+
+	// Replace global tags with local copy
+	tags = localTags
```

Comment:
can you lock only when doing this replacement?

Reply:
fixed in 334a193